src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -288,6 +288,15 @@ protected BoundNode VisitAlways(BoundNode node)
         private BoundNode VisitWithStackGuard(BoundNode node)
         {
             var expression = node as BoundExpression;
+            if (expression == null)
+            {
+                var equalsValue = node as BoundEqualsValue;
+                if (equalsValue != null)
+                {
+                    expression = equalsValue.Value;
```

Comment:
I'm not 100% sure this statement is reachable for C#, but the corresponding logic in VB was necessary.